vllm/model_executor/models/siglip.py

```diff
@@ -489,6 +492,16 @@ def __init__(
         num_hidden_layers_override: Optional[int] = None,
     ):
         super().__init__()
+        if (num_hidden_layers_override is None
+                or num_hidden_layers_override == config.num_hidden_layers):
+            self.need_post_layernorm = True
+        elif num_hidden_layers_override > config.num_hidden_layers:
+            raise ValueError(
+                "num_hidden_layers_override cannot be greater than "
+                "num_hidden_layers")
+        else:
+            self.need_post_layernorm = False
+
```

Comment:
Can we move this logic inside the vision transformer class?

Reply:
I thought so at first, but I don't know how to tell if `post_layernorm` should be skipped in `load_weights`

Reply:
Maybe we can get `need_post_layernorm` in the `SiglipVisionTransformer` variable in the `SiglipVisionModel`

Reply:
> Maybe we can get `need_post_layernorm` in the `SiglipVisionTransformer` variable in the `SiglipVisionModel`
Yeah, that sounds good.